src/thunderbots/software/util/parameter/parameter.h

```diff
@@ -6,128 +6,221 @@
+         *
+         * @param new_value The new value to be set
+         */
+        void setValueInROSParameterServer(T new_value)
+        {
+	    //dynamic reconfigure takes control of setting params //ros::param::set(getROSParameterPath(), new_value);
+        }
+
+        /**
+         * Returns a reference to the Parameter registry. The registry is a list of
+         * pointers to all the existing Parameters.
+         *
+         * @return An immutable reference to the Parameter registry
+         */
+        static const std::vector<std::unique_ptr<Parameter<T>>>& getRegistry()
+        {
+            return Parameter<T>::getMutableRegistry();
+        }
+
+        /**
+         * Returns a reference to the config struct. The config struct contains
+         * all the current configurations
+         *
+         * @return An immutable reference to the Config struct
+         */
+        static const dynamic_reconfigure::Config& getConfigStruct()
         {
-            p->updateValueFromROSParameterServer();
+            return Parameter<T>::getMutableConfigStruct();
         }
-    }
-
-   private:
-    /**
-     * Returns a mutable reference to the Parameter registry. This is the same as the
-     * above getRegistry() function, except that it returns a mutable reference. We need a
-     * mutable version so that member functions can modify the registry. The function is
-     * private so that external sources cannot modify the registry in unexpected ways.
-     *
-     * @return A mutable reference to the Parameter registry
-     */
-    static std::vector<std::unique_ptr<Parameter>>& getMutableRegistry()
-    {
-        static std::vector<std::unique_ptr<Parameter<T>>> instance;
-        return instance;
-    }
-
-    // Store the value so it can be retrieved without fetching from the server again
-    T value_;
-
-    // The global path to the variable in the ROS Parameter Server
-    std::string ros_parameter_path;
+
+        /**
+         * Registers (adds) a Parameter to the registry. Since the unique pointer is moved
+         * into the registry, the pointer may not be accessed by the caller after this
+         * function has been called.
+	 *
+	 * Also registers params to the static configuration struct used to 
+	 * set parameters
+         *
+         * @param parameter A unique pointer to the Parameter to add. This pointer may not
+         * be accessed by the caller after this function has been called.
+         */
+        static void registerParameter(std::unique_ptr<Parameter<T>> parameter)
+        {   
+
+            if constexpr(std::is_same<T, bool>::value){   
+                dynamic_reconfigure::BoolParameter bool_msg;
+
+                bool_msg.name = parameter->name();
+                bool_msg.value = static_cast<bool>(parameter->value());
+
+                Parameter<T>::getMutableConfigStruct().bools.push_back(bool_msg);
+            }
+
+	    else if  constexpr(std::is_same<T, int32_t>::value){   
```

Comment:
Some funky indenting going on here as well.

Reply:
github its rendering it weird, if you open it raw it looks fine, not sure why its happening in all the repos